scripts/segmentation_HALO-20240822a.md

```diff
@@ -0,0 +1,389 @@
+---
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.16.4
+  kernelspec:
+    display_name: Python 3 (ipykernel)
+    language: python
+    name: python3
+---
+
+# Flight segmentation template
+
+a template for flight segmentation developers to work your way through the flight track piece by piece and define segments in time. An EC track and circles are exemplarily shown for 2024-08-13. A YAML file containing the segment time slices as well as optionally specified `kinds`, `name`, `irregularities` or `comments` is generated at the end.
+
+If a flight includes overpasses of a station of the Meteor, you can import and use the function `plot_overpass` from `utils` which will also print the closest time and distance to the target.
+
+```python
+import matplotlib
+import yaml
+import hvplot.xarray
+import xarray as xr
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+from navdata import get_navdata_HALO
+from utils import *
+
+from orcestra.flightplan import bco, mindelo
+cvao = mindelo
+```
+
+```python
+platform = "HALO"
+flight_id = "HALO-20240822a"
+```
+
+## Loading data
+### Get HALO position and attitude
+
+```python
+ds = get_navdata_HALO(flight_id)
+```
+
+### Get dropsonde launch times
+
+```python
+drops = get_sondes_l1(flight_id)
+ds_drops = ds.sel(time=drops, method="nearest")
+```
+
+### Defining takeoff and landing
+On Barbads, the airport runway plus bumps make HALO move between 7.8-8m above WGS84, on Sal between 88.2-88.4m above WGS84. We therefore define the flight time such that altitude must be above 9m on Barbados and 90m on Sal.
+
+```python
+takeoff, landing, duration = get_takeoff_landing(flight_id, ds)
+print("Takeoff time: " + str(takeoff))
+print("Landing time: " + str(landing))
+print(f"Flight duration: {int(duration / 60)}:{int(duration % 60)}")
+```
+
+### Get EC track and EC meeting point
+
+```python
+ec_track = get_ec_track(flight_id, ds)
+dist_ec, t_ec = get_overpass_track(ds, ec_track)
+```
+
+### Get PACE track
+**loading the PACE track for the first time takes 6-7 minutes!**
+Might be worth only if the flight report states a PACE coordination.
```

Comment:
### Get PACE track
**loading the PACE track for the first time takes 6-7 minutes!**
Might be worth only if the flight report states a PACE coordination.
Based on your decision, choose `load_pace = True` or `load_pace = False`!